routing/payment_lifecycle.go

```diff
@@ -597,9 +597,6 @@ func (p *shardHandler) collectResult(attempt *channeldb.HTLCAttemptInfo) (
 
 	case <-p.router.quit:
 		return nil, ErrRouterShuttingDown
-
-	case <-p.quit:
```

Comment:
I think that `p.quit` could be a problem in more places? Specifically `handleResultErr` in `collectResultAsync`?

I think that we could end up in the same bug scenario, just earlier up the callstack?
* multiple shards launched
* shard handler `quit` channel closed (due to permanent failure)
* in `collectResult()` return an error from the switch
* in `handleResultErr` we select on the shard handler's quit channel
* htlc error is not reported

Reply:
Just double checked and I think we're good, because `collectResultAsync` always calls into `collectResult` which may only return prematurely if the router stops. Otherwise we'd always wait for the Switch to report back the result.